### PR TITLE
borrow LTV > 95% liquidation fix

### PR DESCRIPTION
**Pull request by Rishabh42**

### CODE REVIEW

File: test/scriptFlywheel/unit/liquidateTests.spec.ts [hunk 1]
```diff
@@ -952,6 +928,55 @@ contract("SavingAccount.liquidate", async (accounts) => {
...
+                    let UB = new BN(userBorrowVal).mul(new BN(100));
+                    let UD = new BN(getDeposits).mul(new BN(95));
+                    console.log("userBorrowVal", userBorrowVal.toString());
+                    console.log("userBorrowPower", userBorrowPower.toString());
+                    console.log("UD", UD.toString());
+                    console.log("UB", UB.toString());
+
+                    expect(UB).to.be.bignumber.greaterThan(UD);
+
+                    await savingAccount.liquidate(user1, ETH_ADDRESS, addressDAI);
```
Comment (jchittoda) [inline test/scriptFlywheel/unit/liquidateTests.spec.ts:977]: After liquidation the following should happen:
- The liquidator should get the `collateralTokens` 
- The liquidator's token deposit should reduce.
- The borrowers collateralToken should reduce.
Pls see if you can validate the exact number of the above values.

File: test/scriptFlywheel/unit/liquidateTests.spec.ts [hunk 2]
```diff
@@ -952,6 +928,55 @@ contract("SavingAccount.liquidate", async (accounts) => {
...
+                    await mockChainlinkAggregatorforDAI.updateAnswer(originPrice);
```
Comment (jchittoda) [inline test/scriptFlywheel/unit/liquidateTests.spec.ts:978]: Are you checking some other values after this price update? No need to update answer after liquidation, if there is no further checks.

File: test/scriptFlywheel/unit/liquidateTests.spec.ts [hunk 3]
```diff
@@ -952,6 +928,55 @@ contract("SavingAccount.liquidate", async (accounts) => {
...
+                    const getDeposits = await accountsContract.getDepositETH(user1);
+                    const userBorrowPower = await accountsContract.getBorrowPower(user1);
+                    let UB = new BN(userBorrowVal).mul(new BN(100));
+                    let UD = new BN(getDeposits).mul(new BN(95));
+                    console.log("userBorrowVal", userBorrowVal.toString());
+                    console.log("userBorrowPower", userBorrowPower.toString());
+                    console.log("UD", UD.toString());
+                    console.log("UB", UB.toString());
+
+                    expect(UB).to.be.bignumber.greaterThan(UD);
```
Comment (jchittoda) [inline test/scriptFlywheel/unit/liquidateTests.spec.ts:975]: What is the actual percentage between the values? You can calculate the percentage and then ensure that its greater then 95%

**Review by jchittoda — CHANGES_REQUESTED** [review]

File: test/scriptFlywheel/unit/liquidateTests.spec.ts (hunk 2 above)
Comment (Rishabh42) [reply test/scriptFlywheel/unit/liquidateTests.spec.ts]: No. Have removed it

File: test/scriptFlywheel/unit/liquidateTests.spec.ts (hunk 3 above)
Comment (Rishabh42) [reply test/scriptFlywheel/unit/liquidateTests.spec.ts:985]: It's much higher and I have added the check to ensure the LTV > 95%

File: test/scriptFlywheel/unit/liquidateTests.spec.ts (hunk 1 above)
Comment (Rishabh42) [reply test/scriptFlywheel/unit/liquidateTests.spec.ts]: Added checks for these

**Comment by Rishabh42** [comment]
@jchittoda , I have made changes according to your comments, please review 
Thanks

File: test/scriptFlywheel/unit/liquidateTests.spec.ts [hunk 4]
```diff
@@ -154,64 +154,24 @@ contract("SavingAccount.liquidate", async (accounts) => {
                         "The borrower is not liquidatable."
                     );
                 });
-
-                it("when collateral is not sufficient to be liquidated", async function () {
```
Comment (jchittoda) [inline test/scriptFlywheel/unit/liquidateTests.spec.ts:158]: Pls restore this test case if delete unintentionally.

File: test/scriptFlywheel/unit/liquidateTests.spec.ts [hunk 5]
```diff
@@ -743,53 +645,62 @@ contract("SavingAccount.liquidate", async (accounts) => {
...
+                    const userBorrowPower = await accountsContract.getBorrowPower(user2);
+                    let UB = new BN(userBorrowVal).mul(new BN(100));
+                    let UD = new BN(getDeposits).mul(new BN(95));
+                    console.log("userBorrowVal", userBorrowVal.toString());
+                    console.log("userBorrowPower", userBorrowPower.toString());
+                    console.log("UD", UD.toString());
+                    console.log("UB", UB.toString());
+                    expect(UB).to.be.bignumber.greaterThan(UD);
+                    await savingAccount.liquidate(user2, addressDAI, addressUSDC);
+                    await mockChainlinkAggregatorforUSDC.updateAnswer(originPrice);
```
Comment (jchittoda) [inline test/scriptFlywheel/unit/liquidateTests.spec.ts:696]: here also `updateAnswer` is used.

File: test/scriptFlywheel/unit/liquidateTests.spec.ts [hunk 6]
```diff
@@ -743,53 +645,62 @@ contract("SavingAccount.liquidate", async (accounts) => {
...
+                    const getDeposits = await accountsContract.getDepositETH(user2);
+                    const userBorrowPower = await accountsContract.getBorrowPower(user2);
+                    let UB = new BN(userBorrowVal).mul(new BN(100));
+                    let UD = new BN(getDeposits).mul(new BN(95));
+                    console.log("userBorrowVal", userBorrowVal.toString());
+                    console.log("userBorrowPower", userBorrowPower.toString());
+                    console.log("UD", UD.toString());
+                    console.log("UB", UB.toString());
+                    expect(UB).to.be.bignumber.greaterThan(UD);
+                    await savingAccount.liquidate(user2, addressDAI, addressUSDC);
```
Comment (jchittoda) [inline test/scriptFlywheel/unit/liquidateTests.spec.ts:695]: After liquidate, validate that the tokens are given to liquidator.

File: test/scriptFlywheel/unit/liquidateTests.spec.ts (hunk 5 above)
Comment (Rishabh42) [reply test/scriptFlywheel/unit/liquidateTests.spec.ts:696]: Yes, have to remove those. Even I just saw while working on liquidate

**Review by jchittoda — APPROVED** [review]

**Pull request merged**